### ISSUE TITLE
Cannot enable Hyper-V service

### ISSUE THREAD

**Issue by foxlance** [docker/for-win#6714]
- [X] I have tried with the latest version of my channel (Stable or Edge)
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: 62AE401B-9C3F-4738-B9CB-E9CF5B2CAD25/20200514124053

### Expected behavior
I can start Docker Desktop

### Actual behavior
Docker Desktop does not start. I get the error:

**An error occurred**
Cannot enable Hyper-V service
### Information

  - Windows Version: **Win10 Pro. 1903 (OS Build 18362.778)**
  - Docker Desktop Version: **(Stable) 2.3.0.2 (45183)**
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: **No.**

### Steps to reproduce the behavior

  1. Make sure Hyper-V and Containers is ticked on windows Features
  2. Make sure Virtualization is enabled under CPU.
  3. Re-start PC
  4. Start Docker Desktop. Will attempt to start but fail with the aforementioned error.

### Other things to note
- I have **Vagrant** and **VirtualBox** installed. They are not running as a service, nor are they active. I'm not attempting to run these alongside Docker Desktop. I'm simply trying to test Docker Desktop to see if I can finally make the switch. I can use vagrant just fine, so I know Hyper-V and virtualization is enabled and working.

When making the diagnostics, the log file shows the following which may be of interest:
```
Version: 2.3.0.2 (45183)
Channel: stable
Sha1: e9b671cd3cb844cb63ad6fd7c79a4d15424233c4
Started on: 2020/05/14 20:19:28.776
Resources: C:\Program Files\Docker\Docker\resources
OS: Windows 10 Pro
Edition: Professional
Id: 1903
Build: 18362
BuildLabName: 18362.1.amd64fre.19h1_release.190318-1202
File: C:\Users\Lance\AppData\Local\Docker\log.txt
CommandLine: "C:\Program Files\Docker\Docker\Docker Desktop.exe" 
You can send feedback, including this log file, at https://github.com/docker/for-win/issues
[20:19:28.870][GUI               ][Info   ] Starting...
[20:19:28.886][ComponentVersions ][Info   ] Edition community
[20:19:28.886][ComponentVersions ][Info   ] Edition community
[20:19:29.104][AppMigrator       ][Info   ] Current version: 6. Latest version: 6
[20:19:29.126][TrackingSettings  ][Info   ] Crash report and usage statistics are enabled
[20:19:29.131][SegmentApi        ][Info   ] Usage statistic: Identify
[20:19:29.271][SegmentApi        ][Info   ] Usage statistic: appLaunched
[20:19:29.603][ApplicationTemplatesTracking][Info   ] Cannot list templates
[20:19:29.604][SegmentApi        ][Info   ] Usage statistic: eventTemplatesInfo
[20:19:29.605][SegmentApi        ][Info   ] Usage statistic: heartbeat
[20:19:29.668][LoggingMessageHandler][Info   ] [9d5fd0ff] <BackendAPIClient start> GET http://backend/version
[20:19:29.702][LoggingMessageHandler][Info   ] [9d5fd0ff] <BackendAPIClient end> GET http://backend/version -> 200 OK (took 33ms)
[20:19:29.841][LoggingMessageHandler][Info   ] [7fa55231] <BackendAPIClient start> GET http://backend/hyperv/vhdx-size?path=C:%5CProgramData%5CDockerDesktop%5Cvm-data%5CDockerDesktop.vhdx
[20:19:29.997][LoggingMessageHandler][Info   ] [7fa55231] <BackendAPIClient end> GET http://backend/hyperv/vhdx-size?path=C:%5CProgramData%5CDockerDesktop%5Cvm-data%5CDockerDesktop.vhdx -> 200 OK (took 154ms)
[20:19:29.997][LoggingMessageHandler][Info   ] [eeae847a] <BackendAPIClient start> POST http://backend/migrate/app
[20:19:29.997][LoggingMessageHandler][Info   ] [eeae847a] <BackendAPIClient end> POST http://backend/migrate/app -> 204 NoContent (took 1ms)
[20:19:30.013][Engines           ][Debug  ] Starting
[20:19:30.028][LoggingMessageHandler][Info   ] [9c915333] <BackendAPIClient start> POST http://backend/versionpack/enable
[20:19:30.059][LoggingMessageHandler][Info   ] [9c915333] <BackendAPIClient end> POST http://backend/versionpack/enable -> 204 NoContent (took 37ms)
[20:19:30.148][GoBackendProcess  ][Info   ] Starting C:\Program Files\Docker\Docker\resources\com.docker.backend.exe -addr unix:\\.\pipe\dockerBackendApiServer
[20:19:30.154][GoBackendProcess  ][Info   ] Started
[20:19:30.162][MutagenProcess    ][Info   ] Starting C:\ProgramData\Docker\cli-plugins\docker-mutagen.exe mutagen daemon run
[20:19:30.170][MutagenProcess    ][Info   ] Started
[20:19:30.179][EngineStateMachine][Debug  ] sending state Docker.ApiServices.StateMachines.StartTransition to state change sink
[20:19:30.180][EngineStateMachine][Debug  ] State Docker.ApiServices.StateMachines.StartTransition sent to state change sink
[20:19:30.185][EngineStateListener][Debug  ] received state Docker.ApiServices.StateMachines.StartTransition from LinuxHyperV
[20:19:30.191][EngineStateNotificationRecorder][Debug  ] Registered state {"State":"starting","Mode":"linux","date":1589458770}
[20:19:30.192][SystrayNotifications][Info   ] Docker is starting
[20:19:30.196][LoggingMessageHandler][Info   ] [8fc7c3b3] <BackendAPIClient start> POST http://backend/dns/refresh-hosts
[20:19:30.263][LoggingMessageHandler][Info   ] [8fc7c3b3] <BackendAPIClient end> POST http://backend/dns/refresh-hosts -> 204 NoContent (took 67ms)
[20:19:30.266][LinuxHyperVEngine ][Info   ] Stopping watching for host power events
[20:19:30.268][MutagenProcess    ][Info   ] [forwarding] Looking for existing sessions
[20:19:30.268][MutagenProcess    ][Info   ] [forwarding] Session manager initialized
[20:19:30.268][MutagenProcess    ][Info   ] [sync] Looking for existing sessions
[20:19:30.268][MutagenProcess    ][Info   ] [sync] Session manager initialized
[20:19:30.270][LoggingMessageHandler][Info   ] [d37e8907] <BackendAPIClient start> POST http://backend/hyperv/stop
[20:19:30.746][LoggingMessageHandler][Info   ] [02864068] <Server start> GET http://unix/versions
[20:19:30.817][LoggingMessageHandler][Info   ] [02864068] <Server end> GET http://unix/versions -> 200 OK (took 70ms)
[20:19:30.851][GoBackendProcess  ][Info   ] ⇨ http server started on \\.\pipe\dockerVpnKitControl
[20:19:30.852][GoBackendProcess  ][Info   ] started port-forwarding control server on \\\\.\\pipe\\dockerVpnKitControl
[20:19:30.853][GoBackendProcess  ][Info   ] listening on unix:\\\\.\\pipe\\dockerVpnkitData for data connection
[20:19:30.854][GoBackendProcess  ][Info   ] enabling filesystem caching
[20:19:30.854][GoBackendProcess  ][Info   ] filesystem exports are:  (2)
[20:19:30.854][GoBackendProcess  ][Info   ] volume control server listening on \\\\.\\pipe\\dockerVolume
[20:19:30.854][GoBackendProcess  ][Info   ] ⇨ http server started on \\.\pipe\dockerBackendApiServerForGuest
[20:19:30.854][GoBackendProcess  ][Info   ] ⇨ http server started on \\.\pipe\dockerBackendApiServer
[20:19:30.856][GoBackendProcess  ][Info   ] filesystem server listening on 00000000-0000-0000-0000-000000000000:00001003-facb-11e6-bd58-64006a7986d3
[20:19:30.857][GoBackendProcess  ][Info   ] file ownership will be determined by the calling user (\"fake owner\" mode)
[20:19:30.857][GoBackendProcess  ][Info   ] using mfsymlinks
[20:19:31.856][LoggingMessageHandler][Info   ] [d37e8907] <BackendAPIClient end> POST http://backend/hyperv/stop -> 204 NoContent (took 1586ms)
[20:19:31.884][VpnKitBridge      ][Info   ] Starting C:\Program Files\Docker\Docker\resources\vpnkit-bridge.exe --addr listen://00000000-0000-0000-0000-000000000000/000007CF-FACB-11E6-BD58-64006A7986D3 host
[20:19:31.890][VpnKitBridge      ][Info   ] Started
[20:19:31.902][IsoConfig         ][Info   ] Generating CA Cert Bundle
[20:19:31.950][IsoConfig         ][Info   ] CA Cert Bundle Generated
[20:19:31.957][VpnKitBridge      ][Info   ] listening on listen://00000000-0000-0000-0000-000000000000/000007CF-FACB-11E6-BD58-64006A7986D3...
[20:19:31.960][VpnKitBridge      ][Info   ] waiting for connection on listen://00000000-0000-0000-0000-000000000000/000007CF-FACB-11E6-BD58-64006A7986D3...
[20:19:32.057][LoggingMessageHandler][Info   ] [d3dcc17a] <BackendAPIClient start> POST http://backend/hyperv/create
[20:19:32.393][DockerHub         ][Info   ] Login successfull
[20:19:34.198][LoggingMessageHandler][Info   ] [d3dcc17a] <BackendAPIClient end> POST http://backend/hyperv/create -> 204 NoContent (took 2141ms)
[20:19:34.207][VpnKit            ][Info   ] Starting C:\Program Files\Docker\Docker\resources\vpnkit.exe --ethernet \\.\pipe\dockerVpnkit --hosts "" --http "C:\Users\Lance\AppData\Roaming\Docker\http_proxy.json" --gateway-forwards "C:\Users\Lance\AppData\Roaming\Docker\gateway_forwards.json" --diagnostics \\.\pipe\dockerVpnKitDiagnostics --host-names host.docker.internal,docker.for.win.host.internal,docker.for.win.localhost --gateway-names gateway.docker.internal,docker.for.win.gateway.internal,docker.for.win.http.internal --listen-backlog 32 --gateway-ip 192.168.65.1 --host-ip 192.168.65.2 --lowest-ip 192.168.65.3 --highest-ip 192.168.65.14
[20:19:34.214][VpnKit            ][Info   ] Started
[20:19:34.219][LoggingMessageHandler][Info   ] [73abb719] <BackendAPIClient start> POST http://backend/hyperv/start
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: Setting handler to ignore all SIGPIPE signals
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: Version is 218f01482560cba2fa863f9ad872ad51d1e717fc
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: System SOMAXCONN is 2147483647
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: Will use a listen backlog of 32
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: No periodic Gc.compact enabled
[20:19:34.280][VpnKit            ][Error  ] vpnkit.exe: Starting to watch : EINVAL
[20:19:34.280][VpnKit            ][Error  ] vpnkit.exe: Failed to watch hosts file : invalid argument
[20:19:34.280][VpnKit            ][Warning] vpnkit.exe: There is no database: using hardcoded network configuration values
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: DNS server configured with no builtin DNS names; everything will be forwarded
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: 2 upstream DNS servers are configured
[20:19:34.280][VpnKit            ][Error  ] vpnkit.exe: While watching /etc/resolv.conf: ENOENT
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: Disabling transparent HTTP redirection
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: Updating resolvers to use host resolver
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: Secure random number generator is available
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: Add(3): DNS configuration changed to: use host resolver
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: DNS server configured with builtin DNS names [ gateway.docker.internal -> 192.168.65.1, docker.for.win.gateway.internal -> 192.168.65.1, docker.for.win.http.internal -> 192.168.65.1, host.docker.internal -> 192.168.65.2, docker.for.win.host.internal -> 192.168.65.2, docker.for.win.localhost -> 192.168.65.2, vm.internal -> 192.168.65.3 ]
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: Will use the host's DNS resolver
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: Transparent HTTP redirection configuration file C:\Users\Lance\AppData\Roaming\Docker\http_proxy.json has changed
[20:19:34.280][VpnKit            ][Info   ] vpnkit.exe: Reading transparent HTTP redirection from C:\Users\Lance\AppData\Roaming\Docker\http_proxy.json
[20:19:34.281][VpnKit            ][Info   ] vpnkit.exe: Watching transparent HTTP redirection configuration file C:\Users\Lance\AppData\Roaming\Docker\http_proxy.json for changes
[20:19:34.281][VpnKit            ][Info   ] vpnkit.exe: New Gateway forward configuration: []
[20:19:34.281][VpnKit            ][Info   ] vpnkit.exe: Gateway forwards file C:\Users\Lance\AppData\Roaming\Docker\gateway_forwards.json has changed
[20:19:34.281][VpnKit            ][Info   ] vpnkit.exe: Reading gateway forwards file from C:\Users\Lance\AppData\Roaming\Docker\gateway_forwards.json
[20:19:34.281][VpnKit            ][Info   ] vpnkit.exe: Watching gateway forwards file C:\Users\Lance\AppData\Roaming\Docker\gateway_forwards.json for changes
[20:19:34.281][VpnKit            ][Info   ] vpnkit.exe: Configuration server_macaddr = f6:16:36:bc:f9:c6; max_connection = None; dns_path = None; dns = ; resolver = Host; domain = None; allowed_bind_addresses = 0.0.0.0; gateway_ip = 192.168.65.1; host_ip = 192.168.65.2; lowest_ip = 192.168.65.3; highest_ip = 192.168.65.14; dhcp_json_path = None; dhcp_configuration = None; mtu = 1500; http_intercept = None; http_intercept_path = C:\Users\Lance\AppData\Roaming\Docker\http_proxy.json; port_max_idle_time = 300; host_names = host.docker.internal, docker.for.win.host.internal, docker.for.win.localhost; gateway_names = gateway.docker.internal, docker.for.win.gateway.internal, docker.for.win.http.internal; vm_names = vm.internal; udpv4_forwards = []; tcpv4_forwards = []; gateway_forwards_path = C:\Users\Lance\AppData\Roaming\Docker\gateway_forwards.json; pcap_snaplen = 128
[20:19:34.281][VpnKit            ][Error  ] vpnkit.exe: Failed to read gateway forwards from C:\Users\Lance\AppData\Roaming\Docker\gateway_forwards.json: reading C:\Users\Lance\AppData\Roaming\Docker\gateway_forwards.json: No such file or directory.
[20:19:34.281][VpnKit            ][Info   ] vpnkit.exe: New Gateway forward configuration: []
[20:19:34.281][VpnKit            ][Error  ] vpnkit.exe: While watching C:\Users\Lance\AppData\Roaming\Docker\gateway_forwards.json: ENOENT
[20:19:34.281][VpnKit            ][Info   ] vpnkit.exe: Updating transparent HTTP redirection: {
[20:19:34.281][VpnKit            ][Info   ]   "exclude": "",
[20:19:34.281][VpnKit            ][Info   ]   "transparent_http_ports": [
[20:19:34.281][VpnKit            ][Info   ]     80
[20:19:34.281][VpnKit            ][Info   ]   ],
[20:19:34.281][VpnKit            ][Info   ]   "transparent_https_ports": [
[20:19:34.281][VpnKit            ][Info   ]     443
[20:19:34.281][VpnKit            ][Info   ]   ]
[20:19:34.281][VpnKit            ][Info   ] }
[20:19:34.357][LoggingMessageHandler][Info   ] [73abb719] <BackendAPIClient end> POST http://backend/hyperv/start -> 500 InternalServerError (took 137ms)
[20:19:34.361][LinuxHyperVEngine ][Info   ] Stopping watching for host power events
[20:19:34.361][LoggingMessageHandler][Info   ] [99423cac] <BackendAPIClient start> POST http://backend/hyperv/stop
[20:19:35.903][LoggingMessageHandler][Info   ] [99423cac] <BackendAPIClient end> POST http://backend/hyperv/stop -> 204 NoContent (took 1541ms)
[20:19:35.913][VpnKit            ][Info   ] Killing existing vpnkit with PID 14568
[20:19:35.926][VpnKitBridge      ][Info   ] Killing existing vpnkit-bridge with PID 2056


# Here is where it gets interesting:


[20:19:35.940][LoggingMessageHandler][Info   ] [576acce2] <BackendAPIClient start> POST http://backend/windowsfeatures/check
[20:19:36.283][LoggingMessageHandler][Info   ] [576acce2] <BackendAPIClient end> POST http://backend/windowsfeatures/check -> 200 OK (took 342ms)
[20:19:36.289][LoggingMessageHandler][Info   ] [9c998da7] <BackendAPIClient start> GET http://backend/hyperv/check-virtualization
[20:19:36.306][LoggingMessageHandler][Info   ] [9c998da7] <BackendAPIClient end> GET http://backend/hyperv/check-virtualization -> 500 InternalServerError (took 16ms)
[20:19:36.307][EngineStateMachine][Debug  ] sending state Docker.ApiServices.StateMachines.FailedToStartState to state change sink
[20:19:36.307][EngineStateMachine][Debug  ] State Docker.ApiServices.StateMachines.FailedToStartState sent to state change sink
[20:19:36.307][EngineStateListener][Debug  ] received state Docker.ApiServices.StateMachines.FailedToStartState from LinuxHyperV
[20:19:36.308][EngineStateNotificationRecorder][Debug  ] Registered state {"State":"failed to start","Mode":"linux","date":1589458776}
[20:19:36.414][Engines           ][Error  ] Start failed with Cannot enable Hyper-V service
   at Docker.Engines.LinuxHyperVEngine.<DoCheckPreconditionsAsync>d__16.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Desktop\Engines\LinuxHyperVEngine.cs:line 125
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.TaskExtensions.<WrapAsyncInCancellationException>d__0.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\TaskExtensions.cs:line 29
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.StartTransition.<DoRunAsync>d__5.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\StartTransition.cs:line 62
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.ApiServices.StateMachines.EngineStateMachine.<StartAsync>d__14.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.ApiServices\StateMachines\EngineStateMachine.cs:line 72
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Docker.Engines.Engines.<StartAsync>d__24.MoveNext() in C:\workspaces\stable-2.3.x\src\github.com\docker\pinata\win\src\Docker.Desktop\Engines\Engines.cs:line 109)
[20:19:43.878][SegmentApi        ][Info   ] Usage statistic: actionMenuFeedback
[20:19:44.478][LoggingMessageHandler][Info   ] [06d0c2a6] <Server start> POST http://localhost/notifications/subscriptions
[20:19:44.517][LoggingMessageHandler][Info   ] [06e0e664] <NotificationsClient start> GET http://localhost/ping
[20:19:44.563][LoggingMessageHandler][Info   ] [06e0e664] <NotificationsClient end> GET http://localhost/ping -> 200 OK (took 46ms)
[20:19:44.565][LoggingMessageHandler][Info   ] [06d0c2a6] <Server end> POST http://localhost/notifications/subscriptions -> 200 OK (took 86ms)
[20:19:46.425][LoggingMessageHandler][Info   ] [66db606c] <Server start> GET http://localhost/registry/user
[20:19:46.428][LoggingMessageHandler][Info   ] [74cf8665] <Server start> GET http://localhost/docker
[20:19:46.430][LoggingMessageHandler][Info   ] [ebbc8236] <Server start> GET http://localhost/kubernetes
[20:19:46.433][LoggingMessageHandler][Info   ] [ee8d1737] <Server start> GET http://localhost/license/details
[20:19:46.436][LoggingMessageHandler][Info   ] [8f645066] <Server start> GET http://localhost/versions
[20:19:46.436][LoggingMessageHandler][Info   ] [66db606c] <Server end> GET http://localhost/registry/user -> 200 OK (took 10ms)
[20:19:46.436][LoggingMessageHandler][Info   ] [ee8d1737] <Server end> GET http://localhost/license/details -> 200 OK (took 2ms)
[20:19:46.436][LoggingMessageHandler][Info   ] [8f645066] <Server end> GET http://localhost/versions -> 200 OK (took 0ms)
[20:19:46.436][LoggingMessageHandler][Info   ] [74cf8665] <Server end> GET http://localhost/docker -> 200 OK (took 8ms)
[20:19:46.436][LoggingMessageHandler][Info   ] [ebbc8236] <Server end> GET http://localhost/kubernetes -> 200 OK (took 6ms)
[20:19:46.479][GoBackendProcess  ][Info   ] external: GET /system/editor 200 \"axios/0.19.0\" \"\
[20:19:51.712][LoggingMessageHandler][Info   ] [58ce37bd] <Server start> GET http://localhost/desktop/diagnose
[20:19:51.738][LoggingMessageHandler][Info   ] [58ce37bd] <Server end> GET http://localhost/desktop/diagnose -> 204 NoContent (took 26ms)
[20:19:51.738][Diagnostics       ][Warning] Starting to gather diagnostics as User : 'C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe' gather.
[20:19:51.805][GoBackendProcess  ][Info   ] external: GET /events 200 \"Go-http-client/1.1\" \"\
[20:19:51.805][GoBackendProcess  ][Info   ] external: GET /forwards/list 200 \"Go-http-client/1.1\" \"\
[20:19:51.805][GoBackendProcess  ][Info   ] mutagen sync list --output json
[20:19:51.842][GoBackendProcess  ][Info   ] mutagen C:\\ProgramData\\Docker\\cli-plugins\\docker-mutagen.exe mutagen sync list --output json: null\n
[20:19:51.842][GoBackendProcess  ][Info   ] external: GET /sync 200 \"Go-http-client/1.1\" \"\
[20:20:05.160][FeedbackWindow    ][Info   ] Open logs
```

**Comment by Richard-Barrett** [comment]
# Windows
Here is what worked for me:
If you want to do it on Windows I would do the following:
1. Uninstall Everything Related to Docker-Desktop in Control Panel and disable Hyper-V
https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v
2. If you are using Chocolatey you can uninstall all of that there too with `choco uninstall package` and disable Hyper-V https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v
3. Restart your Machine and Enter BIOS Menu and Make sure you have Virtualization and/or Hyperthreading Enabled. 

Use this link to see how to get into BIOS: https://support.lenovo.com/us/en/solutions/ht500006
You may need to reset your BIOS to default (Note not all Machines Support Virtualization) 
4. Allow for normal boot to occur
5. Enable Hyper-V for Windows 10 
https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v
6. Run through the following in an elevated command prompt and allowing for administrative access by right-clicking on command prompt:
```
dism.exe /Online /Enable-Feature:Microsoft-Hyper-V /All
bcdedit /set hypervisorlaunchtype auto
```

7. Restart Your Computer One More Time
8. Boot up
9. You can go to Docker-Desktop or you can install it with Chocolatey
If you want to install it via website, I assume you know where to go. Get the Stable Version. 
10 (Option A). Install as normal via https://www.docker.com/products/docker-desktop
10 (Option B). Use Chocolatey: 
Open up powershell in elevated admin console and install chocolatey
```
Set-ExecutionPolicy Bypass -Scope Process -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
```
https://chocolatey.org/install

Then after you install it run:
```
choco install docker-desktop
```
Chocolatey Link: https://chocolatey.org/packages/docker-desktop#releasenotes
Docker Release Notes: https://docs.docker.com/docker-for-windows/release-notes/

Whatever you decide, you need to make sure 
1. Your System Supports Virtualization
2. You have that enabled in System BIOS
3. Makes sure you have downloaded the WSL 2 background: https://docs.microsoft.com/en-us/windows/wsl/wsl2-kernel

Hope this helps, I too have had so many problems with Docker-Desktop as of late.

**Comment by foxlance** [comment]
@Richard-Barrett Thanks a lot for the swift response! I gave it a couple more attempts yesterday, and tried uninstalling Hyper-V, and Docker-Desktop, to no avail. In fact I ended up installing the Edge version cause I was running out of ideas. Everything on your notes are steps I've already given a try. All except one:

```
bcdedit /set hypervisorlaunchtype auto
```
After that command and a restart, Docker Desktop is now running!

The bad news though is that vagrant doesn't work with that feature turned on, so I have to turn it off and restart if I want to use vagrant. A minor annoyance, but works. Thanks again!
